doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -389,15 +386,13 @@ the propositional implication and function types.
 Applications
 ------------
 
-The expression :token:`term`\ :math:`_0` :token:`term`\ :math:`_1` denotes the
-application of :token:`term`\ :math:`_0` to :token:`term`\ :math:`_1`.
+The expression :n:`@term__0 @term__1` denotes the
+application of :n:`@term__0` to :n:`term__1`.
 
-The expression :token:`term`\ :math:`_0` :token:`term`\ :math:`_1` ...
-:token:`term`\ :math:`_n` denotes the application of the term
-:token:`term`\ :math:`_0` to the arguments :token:`term`\ :math:`_1` ... then
-:token:`term`\ :math:`_n`. It is equivalent to ( … ( :token:`term`\ :math:`_0`
-:token:`term`\ :math:`_1` ) … ) :token:`term`\ :math:`_n` : associativity is to the
-left.
+The expression :n:`term__0 {+ @term__i }` denotes the application of
+the term :n:`@term__0` to the arguments :n:`@term__i`.  It is
+equivalent to :n:`{+ ( } @term__0 @term__1 {+ ) @term__i }`:
+associativity is to the left.
```

Comment:
Yes, parentheses must balance, and indeed I was aware that this isn't reflected in the output. Once again, either we decide that the reader is smart enough to figure it on their own, or we do otherwise. We could indeed keep the `...` but I am not sure how that would be more precise: the reader still has to figure out that the number of repetitions on the right and on the left, marked by two sets of `...` is the same.

Reply:
It's fairly common to see (and not just in Coq) things like:
![image](https://user-images.githubusercontent.com/1253341/71548601-d201f200-2965-11ea-9715-1fea16273073.png)

Whereas our notations give the impression of being a lot more precise (and much less familiar).  What do you think?

Reply:
OK, I buy this argument.